spec/classes/agent_spec.rb

```diff
@@ -122,6 +122,28 @@
         it { is_expected.to contain_file(config_path).with_content %r{^Hostname=test$} }
       end
 
+      context 'configuration file with hostinterfaceitem' do
+        let :params do
+          {
+            hostinterfaceitem: 'system.hostname'
+          }
+        end
+
+        it { is_expected.to contain_file(config_path).with_content %r{^HostInterfaceItem=system.hostname$} }
+      end
+
+      context 'configuration file with hostinterface' do
+        let :params do
+          {
+            hostinterface: 'testinterface',
+            hostinterfaceitem: 'system.hostname'
```

Comment:
Can we only test `hostinterface` here? `hostinterfaceitem` is already tested above. At least as far as I can see, there is no required connection between both parameters.